src/Appwrite/Realtime/Realtime.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Appwrite\Realtime;
+
+use Appwrite\Auth\Auth;
+use Appwrite\Database\Document;
+
+class Realtime
+{
+    /**
+     * @var Document $user
+     */
+    static $user;
+
+    /**
+     * @param Document $user
+     */
+    static function setUser(Document $user)
+    {
+        self::$user = $user;
+    }
+
+    /**
+     * @return array
+     */
+    static function getRoles()
+    {
+        $roles = ['*', 'role:' . ((self::$user->isEmpty()) ? Auth::USER_ROLE_GUEST : Auth::USER_ROLE_MEMBER)];
+        if (!(self::$user->isEmpty())) {
+            $roles[] = 'user:' . self::$user->getId();
+        }
+        foreach (self::$user->getAttribute('memberships', []) as $node) {
+            if (isset($node['teamId']) && isset($node['roles'])) {
+                $roles[] = 'team:' . $node['teamId'];
+
+                foreach ($node['roles'] as $nodeRole) { // Set all team roles
+                    $roles[] = 'team:' . $node['teamId'] . '/' . $nodeRole;
+                }
+            }
+        }
+        return $roles;
+    }
+
+    /**
+     * @param array $channels
+     */
+    static function parseChannels(array $channels)
+    {
+        $channels = array_flip($channels);
+
+        foreach ($channels as $key => $value) {
+            if (strpos($key, 'account.') === 0) {
+                unset($channels[$key]);
+            } elseif ($key === 'account') {
+                if (!empty(self::$user->getId())) {
+                    $channels['account.' . self::$user->getId()] = $value;
+                }
+                unset($channels['account']);
+            }
+        }
+
+        if (\array_key_exists('account', $channels)) {
+            if (self::$user->getId()) {
+                $channels['account.' . self::$user->getId()] = $channels['account'];
+            }
+            unset($channels['account']);
+        }
+
+        return $channels;
+    }
+
+    /**
+     * Identifies the receivers of all subscriptions, based on the permissions and event.
+     * 
+     * @param array $event
+     * @param array $connections
+     * @param array $subscriptions
+     */
+    static function identifyReceivers(array &$event, array &$connections, array &$subscriptions)
+    {
+        $receivers = [];
+        foreach ($connections as $fd => $connection) {
```

Comment:
This iteration might be very bad for performance. It will be better to just use the user roles we have and check if relevant subscription keys exists inside $subscription instead of looping the full list of connections for every event.